pandas/core/indexes/base.py

```diff
@@ -5516,7 +5534,9 @@ def ensure_index_from_sequences(sequences, names=None):
         return MultiIndex.from_arrays(sequences, names=names)
 
 
-def ensure_index(index_like, copy: bool = False):
+def ensure_index(
+    index_like: Union[AnyArrayLike, Sequence], copy: bool = False
```

Comment:
does Sequence not subsume ArrayLike?

Reply:
see https://github.com/pandas-dev/pandas/issues/28770